l1-contracts/src/mock/StakingAssetHandler.sol

```diff
@@ -1,141 +1,129 @@
   event WithdrawerUpdated(address indexed _withdrawer);
-  event AddValidatorPermissionGranted(address indexed _address);
-  event AddValidatorPermissionRevoked(address indexed _address);
 
-  error NotCanAddValidator(address _caller);
-  error NotEnoughTimeSinceLastMint(uint256 _lastMintTimestamp, uint256 _minMintInterval);
+  event UnhingedAdded(address indexed _address);
+  event UnhingedRemoved(address indexed _address);
+
   error CannotMintZeroAmount();
-  error MaxDepositsTooLarge(uint256 _depositAmount, uint256 _maxDepositsPerMint);
+  error ValidatorQuotaFilledUntil(uint256 _timestamp);
 
   function addValidator(address _attester, address _proposer) external;
-  function setRollup(address _rollup) external;
-  function setDepositAmount(uint256 _amount) external;
   function setMintInterval(uint256 _interval) external;
   function setDepositsPerMint(uint256 _depositsPerMint) external;
   function setWithdrawer(address _withdrawer) external;
-  function grantAddValidatorPermission(address _address) external;
-  function revokeAddValidatorPermission(address _address) external;
+  function addUnhinged(address _address) external;
+  function removeUnhinged(address _address) external;
+
+  function getRollup() external view returns (address);
 }
 
 contract StakingAssetHandler is IStakingAssetHandler, Ownable {
   IMintableERC20 public immutable STAKING_ASSET;
+  IRegistry public immutable REGISTRY;
 
-  mapping(address => bool) public canAddValidator;
+  mapping(address => bool) public isUnhinged;
 
-  uint256 public depositAmount;
   uint256 public lastMintTimestamp;
   uint256 public mintInterval;
   uint256 public depositsPerMint;
 
-  IStaking public rollup;
   address public withdrawer;
 
-  modifier onlyCanAddValidator() {
-    require(canAddValidator[msg.sender], NotCanAddValidator(msg.sender));
-    _;
-  }
-
   constructor(
     address _owner,
     address _stakingAsset,
-    address _rollup,
+    IRegistry _registry,
     address _withdrawer,
-    uint256 _depositAmount,
     uint256 _mintInterval,
     uint256 _depositsPerMint,
-    address[] memory _canAddValidator
+    address[] memory _unhinged
   ) Ownable(_owner) {
     require(_depositsPerMint > 0, CannotMintZeroAmount());
 
     STAKING_ASSET = IMintableERC20(_stakingAsset);
-
-    rollup = IStaking(_rollup);
-    emit RollupUpdated(_rollup);
+    REGISTRY = _registry;
 
     withdrawer = _withdrawer;
     emit WithdrawerUpdated(_withdrawer);
 
-    depositAmount = _depositAmount;
-    emit DepositAmountUpdated(_depositAmount);
-
     mintInterval = _mintInterval;
     emit IntervalUpdated(_mintInterval);
 
     depositsPerMint = _depositsPerMint;
     emit DepositsPerMintUpdated(_depositsPerMint);
 
-    for (uint256 i = 0; i < _canAddValidator.length; i++) {
-      canAddValidator[_canAddValidator[i]] = true;
-      emit AddValidatorPermissionGranted(_canAddValidator[i]);
+    for (uint256 i = 0; i < _unhinged.length; i++) {
+      isUnhinged[_unhinged[i]] = true;
```

Comment:
these names man

Reply:
It is the only fun I am allowed to have now 😭